src/ledger/fuzz.zig

```diff
@@ -0,0 +1,330 @@
+        count,
+        max_actions,
+    );
+}
+
+fn dbDelete(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.delete",
+        BlockstoreDB.delete,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbDeleteFilesInRange(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const start = random.int(u32);
+    const end = blk: {
+        const end_ = random.int(u32);
+        if (end_ < start)
+            break :blk (end_ +| start)
+        else
+            break :blk end_;
+    };
+
+    try performDbAction(
+        "RocksDb.deleteFilesInRange",
+        BlockstoreDB.deleteFilesInRange,
+        .{ db, cf1, start, end },
+        count,
+        max_actions,
+    );
+}
+
+fn dbGetBytes(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.getBytes",
+        BlockstoreDB.getBytes,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbGet(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.get",
+        BlockstoreDB.get,
+        .{ db, allocator, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+fn dbCount(
+    db: *BlockstoreDB,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    try performDbAction(
+        "RocksDb.count",
+        BlockstoreDB.count,
+        .{ db, cf1 },
+        count,
+        max_actions,
+    );
+}
+
+fn dbContains(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    const key = random.int(u32);
+    try performDbAction(
+        "RocksDb.contains",
+        BlockstoreDB.contains,
+        .{ db, cf1, key },
+        count,
+        max_actions,
+    );
+}
+
+// Batch API
+fn batchDeleteRange(
+    db: *BlockstoreDB,
+    random: *const std.rand.Random,
+    count: *std.atomic.Value(u64),
+    max_actions: ?usize,
+) !void {
+    var last_print_msg_count: u64 = 0;
+    while (true) {
+        if (max_actions) |max| {
+            if (count.load(.monotonic) >= max) {
+                std.debug.print("Batch actions reached max actions: {}\n", .{max});
+                break;
+            }
+        }
+        const start = random.int(u32);
+        const end = blk: {
+            const end_ = random.int(u32);
+            if (end_ < start)
+                break :blk (end_ +| start)
+            else
+                break :blk end_;
+        };
```

Comment:
same here

Reply:
Applied as part of the commit that checks the methods return expected output

https://github.com/Syndica/sig/pull/452/commits/9e06b51485f7c019d6d2404e4ade42e270ac008a